code/datums/quirks/positive_quirks.dm

```diff
@@ -362,3 +362,165 @@
+		deltimer(recovering_timer)
+		recovering_timer = null
+
+	if(skip_timers)
+		on_planet_for_too_long(afflicted, TRUE)
+	else
+		var/exercise_bonus = afflicted.has_status_effect(/datum/status_effect/exercised) ? 2 : 1
+		planetside_timer = addtimer(CALLBACK(src, PROC_REF(on_planet_for_too_long), afflicted), planet_period * exercise_bonus, TIMER_UNIQUE | TIMER_STOPPABLE)
+		afflicted.add_mood_event("spacer", /datum/mood_event/spacer/on_planet)
+		afflicted.add_movespeed_modifier(/datum/movespeed_modifier/spacer/on_planet)
+		afflicted.remove_status_effect(/datum/status_effect/spacer) // removes the wellness effect.
+		to_chat(afflicted, span_danger("You feel a bit sick under the gravity here."))
+
+/**
+ * Ran after remaining on a planet for too long.
+ *
+ * * afflicted - the mob arriving / same as quirk holder
+ * * skip_timers - if TRUE, this is being done instantly / should not have feedback (such as in init)
+ */
+/datum/quirk/spacer_born/proc/on_planet_for_too_long(mob/living/afflicted, skip_timers = FALSE)
+	if(QDELETED(src) || QDELETED(afflicted))
+		return
+
+	planetside_timer = null
+	afflicted.apply_status_effect(/datum/status_effect/spacer/gravity_sickness)
+	afflicted.add_mood_event("spacer", /datum/mood_event/spacer/on_planet/too_long)
+	afflicted.add_movespeed_modifier(/datum/movespeed_modifier/spacer/on_planet/too_long)
+	if(!skip_timers)
+		to_chat(afflicted, span_danger("You've been here for too long. The gravity really starts getting to you."))
+
+// Going back into space
+
+/**
+ * Ran when returning to space / somewhere with low gravity.
+ *
+ * * afflicted - the mob arriving / same as quirk holder
+ * * skip_timers - if TRUE, this is being done instantly / should not have feedback (such as in init)
+ */
+/datum/quirk/spacer_born/proc/in_space(mob/living/afflicted, skip_timers = FALSE)
+	if(planetside_timer)
+		deltimer(planetside_timer)
+		planetside_timer = null
+
+	if(skip_timers)
+		comfortably_in_space(afflicted, TRUE)
+	else
+		recovering_timer = addtimer(CALLBACK(src, PROC_REF(comfortably_in_space), afflicted), recover_period, TIMER_UNIQUE | TIMER_STOPPABLE)
+		afflicted.remove_status_effect(/datum/status_effect/spacer) // removes the sickness effect, but not the other modifiers.
+		to_chat(afflicted, span_green("You start feeling better now that you're back in space."))
+
+/**
+ * Ran when living back in space for a long enough period.
+ *
+ * * afflicted - the mob arriving / same as quirk holder
+ * * skip_timers - if TRUE, this is being done instantly / should not have feedback (such as in init)
+ */
+/datum/quirk/spacer_born/proc/comfortably_in_space(mob/living/afflicted, skip_timers = FALSE)
+	if(QDELETED(src) || QDELETED(afflicted))
+		return
+
+	recovering_timer = null
+	afflicted.apply_status_effect(/datum/status_effect/spacer/gravity_wellness)
+	afflicted.add_mood_event("spacer", /datum/mood_event/spacer/in_space)
+	afflicted.add_movespeed_modifier(/datum/movespeed_modifier/spacer/in_space)
```

Comment:
Do you think this should modify drift speed?

Reply:
I guess just the quirk itself not space acclimating or anything

Reply:
I did want it to speed up drifting but I didn't look into how, guess I should do that